source/cppfront.cpp

```diff
@@ -3517,18 +3518,23 @@ class cppfront
             else
             {
                 //  For the first operator only, if we are emitting a "_ =" discard
-                //  then we've already emitted the cast to void and don't need the =
-                if (suppress_operator) {
-                    assert( x.op->type() == lexeme::Assignment );
-                    suppress_operator = false;
-                }
-                else {
+                //  then we don't need the =
+                if (
+                    !emit_discard
+                    || !first
+                ) {
                     printer.print_cpp2(" ", n.position());
                     emit(*x.op);
+                    printer.print_cpp2(" ", n.position());
                 }
-                printer.print_cpp2(" ", n.position());
                 emit(*x.expr);
```

Comment:
I think this line just needs to be changed to the following to resolve the merge conflict, correct?

```
                printer.print_cpp2(" ", n.position());
                
                //  When assigning a single expression-list, we can
                //  take over direct control of emitting it without needing to
                //  go through the whole grammar, and surround it with braces
                if (
                    x.op->type() == lexeme::Assignment
                    && x.expr->is_expression_list()
                    )
                {
                    printer.print_cpp2( "{ ", n.position() );
                    emit(*x.expr->get_expression_list(), false);
                    printer.print_cpp2( " }", n.position() );
                }
                //  Otherwise, just emit the general expression as usual
                else {
                    emit(*x.expr);
                }
```

Reply:
Almost, without the first line.